### ISSUE TITLE
Route validation is overly aggressive

### ISSUE THREAD

**Issue by cnorthwood** [cloudtools/troposphere#792]
Hi,

Since updating to 1.9.5, our previously valid template now errors. I tracked it down to this commit: https://github.com/cloudtools/troposphere/commit/25c40f6c5b40dd5ecbef756475c63c949deb4c31

Our code looks a bit like this:

```
Route(
                    pri_route,
                    DestinationCidrBlock=EVERYWHERE,
                    RouteTableId=Ref(pri_route_table),
                    InstanceId=If(
                        USE_NAT_GATEWAY,
                        Ref('AWS::NoValue'),
                        Ref(NAT)
                    ),
                    NatGatewayId=If(
                        USE_NAT_GATEWAY,
                        Ref(NAT_GATEWAY),
                        Ref('AWS::NoValue')
                    )
                )
```

Because InstanceId and NatGatewayId are specified, validation fails despite the resulting template being otherwise valid (because of the ifs). I'm wondering if the correct thing to do is to roll back the validation?

/cc @tigertoes

**Comment by tigertoes** [comment]
This is a duplicate of #778, which was fixed in #780 (and is now in HEAD).

**Comment by cnorthwood** [comment]
Aha, must have missed those. Thanks.
